.github/workflows/test.yml

```diff
@@ -3,15 +3,50 @@ name: codacy-coverage-reporter
 on: push
 
 jobs:
-  install-and-test:
-    name: install and test
+  test:
+    name: setup environment and run tests
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
-      - name: npm install
+      - name: Check out repository code
+        uses: actions/checkout@master
+      - name: Node Setup
         uses: actions/setup-node@v2
         with:
           node-version: 14
       - run: npm install
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.17
+      - name: Install latest version of Kind
+        run: go install sigs.k8s.io/kind@v0.17.0
+      - name: Update PATH
+        run: PATH=$(go env GOPATH)/bin:$PATH
+      - name: Create Kind cluster
+        run: kind create cluster --name kind-cluster --config=worker-nodes-kind.yml
+      - name: Create OpenFaaS namespaces
+        run: kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
+      - name: Set up Helm
+        uses: azure/setup-helm@v3
+        with:
+          version: v3.10.0
+      - name: Add OpenFaaS Helm Chart
+        run: helm repo add openfaas https://openfaas.github.io/faas-netes/
+      - name: Deploy OpenFaaS Helm Chart
+        run: |
+          helm repo update \
+            && helm upgrade openfaas --install openfaas/openfaas \
+            --namespace openfaas  \
+            --set functionNamespace=openfaas-fn \
+            --set basic_auth=false \
+            --set generateBasicAuth=false \
+            --set ceScaling=true \
+            --set openfaasImagePullPolicy=IfNotPresent
+      - name: Wait for deployment "gateway" rollout
+        run: kubectl rollout status deploy/gateway --timeout 120s -n openfaas
+      - run: kubectl get events -n openfaas
+      - run: kubectl get pods -n openfaas
+      - name: Port Forwarding to port on gateway-external
+        run: kubectl port-forward service/gateway-external 8080:8080 -n openfaas &
```

Comment:
Would it make sense to add this to a Makefile or package.json, so we run this quickly locally?